server.js

```diff
@@ -2,37 +2,162 @@ import express from 'express'
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const path = require('path')
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.sendFile(path.join(__dirname+'/public/index.html'))
 })
 
-// Start the server
+app.get('/shows', async (req, res) => {   
+  const { title, year, country, actor } = req.query
+  let myFilter = {}
+
+  if (title) {
+    myFilter['title'] = new RegExp(`${title}`, 'i')
+  }
+  
+  if (year) {
+    myFilter['release_year'] = year;
+  }
+
+  if (country) {
+    myFilter['country'] = new RegExp(`${country}`, 'i')
+  }
+
+  if (actor) {
+    myFilter['cast'] = new RegExp(`${actor}`, 'i')
+  }
+  
+  const show = await Shows.find(myFilter).collation({ locale: 'en_US', strength: 1 });
+  
+  const page = +(req.query.page ?? 1)
+  const pageSize = +(req.query.pageSize ?? 20)
+  const startIndex = (page -1) * pageSize
+  const endIndex = startIndex + pageSize 
+  const filteredShows = show.slice(startIndex, endIndex)
+  const returnObject = { 
+    pageSize: pageSize,
+    page: page,
+    maxPages: Math.ceil(show.length/pageSize),
+    totalShows: show.length,
+    results: filteredShows
+  }
+
+  if (returnObject.results.length === 0) {
+    res.status(204).json("No results to show")
+  } else {
+    res.json(returnObject)
+  }
+})
+
+app.get('/shows/:id', async (req, res) => {
+  
+  const show = await Shows.findOne({ show_id: req.params.id })
+  
+  if (show) {
+    res.json(show)
+  } else {
+    res.status(404).json({ error: `This show does not exist!`})
+  }
+})
+
+app.get('/categories', async (req, res) => {
+  let categories = []
+
+  const show = await Shows.find()
+  
+  await show.forEach((item) => {
+    if (!categories.includes(item.type)) {
+      categories.push(item.type)
+    } else return
+  })
+
+  res.json(categories)
+
+})
+
+app.get('/categories/:category', async (req, res) => {
+  const { category } = req.params
+
+  const categoryFind = await Shows.find( { type: category } ).collation({ locale: 'en_US', strength: 1 })
+  
+  const page = +(req.query.page ?? 1)
+  const pageSize = +(req.query.pageSize ?? 20)
+  const startIndex = (page -1) * pageSize
+  const endIndex = startIndex + pageSize 
+  const filteredCategory = categoryFind.slice(startIndex, endIndex)
+  const returnObject = { 
+    pageSize: pageSize,
+    page: page,
+    maxPages: Math.ceil(categoryFind.length/pageSize),
+    totalShows: categoryFind.length,
+    results: filteredCategory
```

Comment:
You repeated this block of code, so if you still want to deliver pagination this way then the good thing to do would be extracting it to separate function and jus call it when needed :)